addons/report/models/report.py

```
@@ -76,6 +76,147 @@ def _get_wkhtmltopdf_bin():
+    documents_paths = []
+    temporary_files = []
+
+    for index, reporthtml in enumerate(bodies):
+        local_command_args = []
+        pdfreport_fd, pdfreport_path = tempfile.mkstemp(suffix='.pdf', prefix='report.tmp.')
+
+        # Wkhtmltopdf handles header/footer as separate pages. Create them if necessary.
+        if headers:
+            head_file_fd, head_file_path = tempfile.mkstemp(suffix='.html', prefix='report.header.tmp.')
+            temporary_files.append(head_file_path)
+            with closing(os.fdopen(head_file_fd, 'w')) as head_file:
+                head_file.write(headers[index])
+            local_command_args.extend(['--header-html', head_file_path])
+        if footers:
+            foot_file_fd, foot_file_path = tempfile.mkstemp(suffix='.html', prefix='report.footer.tmp.')
+            temporary_files.append(foot_file_path)
+            with closing(os.fdopen(foot_file_fd, 'w')) as foot_file:
+                foot_file.write(footers[index])
+            local_command_args.extend(['--footer-html', foot_file_path])
+
+        # Body stuff
+        content_file_fd, content_file_path = tempfile.mkstemp(suffix='.html', prefix='report.body.tmp.')
+        temporary_files.append(content_file_path)
+        with closing(os.fdopen(content_file_fd, 'w')) as content_file:
+            content_file.write(reporthtml)
+
+        try:
+            wkhtmltopdf = [_get_wkhtmltopdf_bin()] + command_args + local_command_args
+            wkhtmltopdf += [content_file_path] + [pdfreport_path]
+            process = subprocess.Popen(wkhtmltopdf, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            out, err = process.communicate()
+
+            if process.returncode not in [0, 1]:
+                raise UserError(_('Wkhtmltopdf failed (error code: %s). '
+                                  'Message: %s') % (str(process.returncode), err))
+
+            documents_paths.append(pdfreport_path)
+        except:
+            raise
```

Can you raise an explicit error or log it¿?